### PR TITLE
Adding a goal boxes test case: txn group

### PR DESCRIPTION
**Pull request by tzaffi**
## Summary

As I was writing an E2E test for boxes in indexer, I ran into a problem that seemed like it could be a bug with setting boxes in the same transaction group in which they were created. As a sanity check, I extended `test/scripts/e2e_subs/box-search.sh` with an atomic group that creates and sets a box. 

Upshot: it passed with flying colors.

I think its a useful test case, and also provides a more recent example of grouping transactions together in goal.

### Additional Changes
* Minor tweak of `cmd/goal/README.md`
* Updated e2e subs py-algorand-sdk dependency to [latest release 1.16.0](https://pypi.org/project/py-algorand-sdk/)

## Test Plan

CI

### CODE REVIEW

File: test/scripts/e2e_subs/box-search.sh [hunk 1]
```diff
@@ -79,4 +79,40 @@ BOX_LIST=$(${gcmd} app box list --app-id "$APPID" --max 1)
...
+
+# Set:
+BOX_VALUE="str:I'm a wonderful box"
+${gcmd} app call --from "$ACCOUNT" --app-id "$APPID" --box "$BOX_NAME" --app-arg "str:set" --app-arg "$BOX_NAME" --app-arg "str:$BOX_VALUE" -o box_set.txn
+
+# Group them, sign and broadcast:
+cat box_create.txn box_set.txn > create_n_set.txn
+${gcmd} clerk group -i create_n_set.txn -o group.txn
+${gcmd} clerk sign -i group.txn -o group.stx
+${gcmd} clerk rawsend -f group.stx
```
Comment (tzaffi) [inline test/scripts/e2e_subs/box-search.sh:96]:
```suggestion
${gcmd} clerk group -i create_n_set.txn -o box_group.txn
${gcmd} clerk sign -i group.txn -o box_group.stx
${gcmd} clerk rawsend -f box_group.stx
```

File: test/scripts/e2e_subs/box-search.sh [hunk 2]
```diff
@@ -79,4 +79,40 @@ BOX_LIST=$(${gcmd} app box list --app-id "$APPID" --max 1)
...
+${gcmd} app call --from "$ACCOUNT" --app-id "$APPID" --box "$BOX_NAME" --app-arg "str:create" --app-arg "$BOX_NAME" -o box_create.txn
+
+# Set:
+BOX_VALUE="str:I'm a wonderful box"
+${gcmd} app call --from "$ACCOUNT" --app-id "$APPID" --box "$BOX_NAME" --app-arg "str:set" --app-arg "$BOX_NAME" --app-arg "str:$BOX_VALUE" -o box_set.txn
+
+# Group them, sign and broadcast:
+cat box_create.txn box_set.txn > create_n_set.txn
+${gcmd} clerk group -i create_n_set.txn -o box_group.txn
+${gcmd} clerk sign -i group.txn -o box_group.stx
```
Comment (tzaffi) [inline test/scripts/e2e_subs/box-search.sh:95]:
```suggestion
${gcmd} clerk sign -i box_group.txn -o box_group.stx
```

File: test/scripts/e2e_subs/box-search.sh [hunk 3]
```diff
@@ -79,4 +79,40 @@ BOX_LIST=$(${gcmd} app box list --app-id "$APPID" --max 1)
...
+${gcmd} app call --from "$ACCOUNT" --app-id "$APPID" --box "$BOX_NAME" --app-arg "str:create" --app-arg "$BOX_NAME" -o box_create.txn
+
+# Set:
+BOX_VALUE="str:I'm a wonderful box"
+${gcmd} app call --from "$ACCOUNT" --app-id "$APPID" --box "$BOX_NAME" --app-arg "str:set" --app-arg "$BOX_NAME" --app-arg "str:$BOX_VALUE" -o box_set.txn
+
+# Group them, sign and broadcast:
+cat box_create.txn box_set.txn > create_n_set.txn
+${gcmd} clerk group -i create_n_set.txn -o box_group.txn
+${gcmd} clerk sign -i box_group.txn -o box_group.stx
```
Comment (tzaffi) [inline test/scripts/e2e_subs/box-search.sh:95]:
```suggestion
cat box_create.txn box_set.txn > box_create_n_set.txn
${gcmd} clerk group -i box_create_n_set.txn -o box_group.txn
${gcmd} clerk sign -i box_group.txn -o box_group.stx
```

File: test/scripts/e2e_subs/box-search.sh [hunk 4]
```diff
@@ -79,4 +79,40 @@ BOX_LIST=$(${gcmd} app box list --app-id "$APPID" --max 1)
...
 
+# Create and set a box in an atommic txn group:
+
+# Create:
+BOX_NAME="str:great box"
+${gcmd} app call --from "$ACCOUNT" --app-id "$APPID" --box "$BOX_NAME" --app-arg "str:create" --app-arg "$BOX_NAME" -o box_create.txn
+
+# Set:
+BOX_VALUE="str:I'm a wonderful box"
+${gcmd} app call --from "$ACCOUNT" --app-id "$APPID" --box "$BOX_NAME" --app-arg "str:set" --app-arg "$BOX_NAME" --app-arg "str:$BOX_VALUE" -o box_set.txn
```
Comment (jannotti) [inline test/scripts/e2e_subs/box-search.sh:90]: Perhaps it would be good to try this without the `--box` in the second tx. Should still be fine.

**Review by jannotti — APPROVED** [review]
Seems useful. Maybe extend it a bit.

**Pull request merged**